From 00d9621d6cd3f42bc92e5a095a780e77bc0f34df Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Wed, 8 Sep 2021 09:35:03 +0100
Subject: [PATCH 08/19] local: Fix detection of scan-element channels

Previously, the detection of buffer-capable channels would fail when the
channel had attributes outside the scan_elements/ folder, as the channel
would be correctly marked as buffer-capable when the first attribute
was found, but then overriden as non-buffer-capable as soon as an
attribute outside the scan_elements/ folder was found.

It went unnoticed until now, most likely because there is generally
either scan-elements attributes or non-scan-elements attributes, and
rarely both.

Fixes #740.

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 local.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/local.c b/local.c
index bbb62fea..e9095201 100644
--- a/local.c
+++ b/local.c
@@ -1476,7 +1476,7 @@ static int add_channel(struct iio_device *dev, const char *name,
 			free(channel_id);
 			ret = add_attr_to_channel(chn, name, path,
 					dir_is_scan_elements);
-			chn->is_scan_element = dir_is_scan_elements && !ret;
+			chn->is_scan_element |= dir_is_scan_elements && !ret;
 			return ret;
 		}
 	}
-- 
2.30.2

